lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3269,32 +3269,39 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
 } // namespace
 
 namespace {
-class ConvertAtenOnesOp : public OpConversionPattern<AtenOnesOp> {
-public:
-  using OpConversionPattern::OpConversionPattern;
+// Converts AtenOnesOp and AtenZerosOp
+struct ConvertAtenOnesZerosOp : ConversionPattern {
+  ConvertAtenOnesZerosOp(TypeConverter &typeConverter, MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+
   LogicalResult
-  matchAndRewrite(AtenOnesOp op, OpAdaptor adaptor,
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
+    if (!isa<AtenOnesOp, AtenZerosOp>(op))
+      return rewriter.notifyMatchFailure(op,
+                                         "not a supported ones or zeros op");
+
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
-    Location loc = op.getLoc();
+    Location loc = op->getLoc();
 
     // We ignore device, but add simple asserts for unimplemented kwargs
-    if (!op.layout().getType().isa<Torch::NoneType>())
+    if (!operands[2].getType().isa<Torch::NoneType>())
```

Comment:
This might be the cause of the error you are seeing. Although it may seem like the `operands` array are equivalent to the arguments one would get by doing `op.argumentName()`, they are actually different! The `operands` array represent the arguments to the operation but with types converted to the types expected by the next dialect. For example, if your op takes as an argument a `!torch.vtensor<[1,2],f32>`, the corresponding argument in the operands list will have type `!tensor<1x2xf32>`. On the other hand, the operands that you get by doing `op.argumentName()` are still at the torch dialect level, so their types are `torch` types.

In this particular case, because you are comparing to a `torch` type, you should be using the argument by doing `op.argumentName()`. The same applies for every time you pattern match below.

Rules of thumb:

Use `op.argumentName()` when:
- you're extracting values or checking if something is a `torch` type

Use `operands` array when:
- you're extracting type information expecting a tensor type (such as when you calculate rank or result element type and you cast to `RankedTensorType` along the way)
- you want your operand to continue living after the `TorchToLinalg` conversion is over (This is what ops that modify tensors do, they pass an element of the `operands` list to the `linalg.generic` calculation because the tensor should still exist after the conversion is over)

Reply:
Thanks, but `op` here is a pointer to the AtenOp, due to which we can't do `op.argumentName()`. Please correct me if I'm wrong.

Reply:
That's correct. There are two approaches that come to mind for handling this. One is to do what the `ConvertReductionOp` does, which is to dynamic cast the op and then extract the information that is specific to the op, as well as define information that is specific for the op (such as the element value used to fill the tensor). 

The other approach, if `aten.ones` and `aten.zeros` had the exact same interface, would be to use a template. However, because here you have code that requires dynamic knowledge to determine the element value used to fill the tensor, I would go with the first approach.

Reply:
Actually, a template is also a viable option because you can pass the fill value as a template argument:

```cpp
template<typename OpTy, int FillVal>
```

Feel free to choose whichever you find more appropriate

Reply:
A third option is to keep the conversion classes separate, but write a helper function for the common calculations that both op-converters can share